backend/common/src/main/java/ai/verta/modeldb/common/nats/JetstreamConnector.java

```diff
@@ -3,39 +3,66 @@
 import ai.verta.modeldb.common.exceptions.ModelDBException;
 import io.nats.client.*;
 import io.nats.client.api.*;
+import io.nats.client.impl.Headers;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import io.opentelemetry.context.propagation.TextMapSetter;
 import java.io.IOException;
 import java.time.Duration;
-import java.util.Collection;
-import java.util.HashMap;
-import java.util.Map;
-import java.util.Optional;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
 import java.util.function.Supplier;
+import javax.annotation.Nullable;
 import lombok.extern.log4j.Log4j2;
 import org.springframework.beans.factory.DisposableBean;
 
 @Log4j2
 public class JetstreamConnector implements DisposableBean {
+  static final AttributeKey<String> SUBJECT_ATTRIBUTE_KEY = AttributeKey.stringKey("subject");
+  static final AttributeKey<String> STREAM_NAME_ATTRIBUTE_KEY =
+      AttributeKey.stringKey("streamName");
+  static final AttributeKey<String> DURABLE_NAME_ATTRIBUTE_KEY = AttributeKey.stringKey("durable");
+  static final AttributeKey<String> DELIVER_GROUP_NAME_KEY =
+      AttributeKey.stringKey("deliverGroupName");
   private static final Duration RECONNECT_WAIT_INTERVAL = Duration.ofSeconds(2L);
   private static final int MEGABYTE = 1024 * 1024;
   private static final int MAX_STREAM_BYTES = 100 * MEGABYTE;
   private static final int MAX_STREAM_MESSAGES = 10000;
+  private static final JetstreamMessageHeaderHandler JETSTREAM_MESSAGE_HEADER_HANDLER =
+      new JetstreamMessageHeaderHandler();
   private final int maxMessageReplicas;
   private final Map<String, JetStream> jetStreamMap = new HashMap<>();
   private final JetstreamConfig config;
   private final Supplier<Collection<String>> streamNamesToManage;
   private Connection natsConnection;
+  private final OpenTelemetry openTelemetry;
 
   public JetstreamConnector(
       JetstreamConfig config, Supplier<Collection<String>> streamNamesToManage) {
+    this(config, streamNamesToManage, OpenTelemetry.noop());
```

Comment:
Would we still use the noop until everything it set in place both microservices?

Reply:
yes, exactly.